src/cli/commands/force/lightning/lwc/test/ui/configure/mobile.ts

```diff
@@ -0,0 +1,453 @@
+
+        const baseUrl = CommandLineUtils.resolveFlag(
+            this.flags.baseurl,
+            Mobile.defaultTestRunnerBaseUrl
+        );
+
+        const injectionConfigsPath = CommandLineUtils.resolveFlag(
+            this.flags.injectionconfigs,
+            ''
+        );
+
+        return RequirementProcessor.execute(this.commandRequirements)
+            .then(async () => {
+                // execute the create command
+                this.logger.info(
+                    'Setup requirements met, continuing with Mobile UI Test Configuration command'
+                );
+
+                const device:
+                    | AndroidVirtualDevice
+                    | IOSSimulatorDevice
+                    | null
+                    | undefined = isAndroid
+                    ? await AndroidUtils.fetchEmulator(deviceName)
+                    : await IOSUtils.getSimulator(deviceName);
+                if (!device) {
+                    return Promise.reject(
+                        new SfError(
+                            util.format(
+                                messages.getMessage(
+                                    'error:deviceNotFoundDescription'
+                                ),
+                                deviceName
+                            )
+                        )
+                    );
+                }
+
+                return this.executeCreateConfigFile(
+                    isAndroid,
+                    device,
+                    output,
+                    testFramework,
+                    bundlePath,
+                    appActivity,
+                    appPackage,
+                    runner,
+                    port,
+                    baseUrl,
+                    injectionConfigsPath
+                );
+            })
+            .then(() => {
+                this.logger.info(`Config file created at ${output}`);
+                return Promise.resolve();
+            })
+            .catch((error) => {
+                this.logger.warn(`Failed to created config file - ${error}`);
+                return Promise.reject(error);
+            });
+    }
+
+    // The following is public for unit testing purposes only
+    public executeCreateConfigFile(
+        isAndroid: boolean,
+        device: AndroidVirtualDevice | IOSSimulatorDevice,
+        output: string,
+        testFramework: string,
+        bundlePath: string,
+        appActivity: string,
+        appPackage: string,
+        runner: string,
+        port: string,
+        baseUrl: string,
+        injectionConfigsPath: string
+    ): Promise<void> {
+        // get the version by taking the last part of the runtimeId
+        // eg: iOS 16.2 => 16.2
+        const iOSVersion = (
+            (device as IOSSimulatorDevice | null)?.runtimeId ?? ''
+        )
+            .split(' ')
+            .pop();
+
+        const iOSCapabilities = dedent`
```

Comment:
TBH I'm not sure how I feel about inlining these configurations for expansion, vs. (admittedly with some additional effort) maintaining these baseline templates as self-contained template files somewhere in the package, which we would read in and replace tokens with the proper variable values.

As a developer, I typically find it easier after the fact to peruse the shape of these kinds of configurations from their own isolated context, rather than have to chase down their root configuration in a code path somewhere. Especially when the end configuration's common format is in a self-contained configuration file. What do you think?